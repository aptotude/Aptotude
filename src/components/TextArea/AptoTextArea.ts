export { TextArea as AptoTextArea } from "semantic-ui-react";
import "semantic-ui-css/semantic.min.css";
