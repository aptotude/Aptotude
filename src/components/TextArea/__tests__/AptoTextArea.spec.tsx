import React from "react";
import { render, fireEvent } from "react-testing-library";
import { AptoTextArea } from "../AptoTextArea";

describe("AptoTextArea", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<AptoTextArea />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("calls onChange", () => {
    const onChangeMock = jest.fn();
    const placeholderText = "Full Name";
    const { getByPlaceholderText } = render(
      <AptoTextArea onChange={onChangeMock} placeholder={placeholderText} />
    );
    const input = getByPlaceholderText(placeholderText);
    const newText = "Jane Doe";
    fireEvent.change(input, { target: { value: newText } });

    expect(onChangeMock.mock.calls[0][1].value).toBe(newText);
  });
});
