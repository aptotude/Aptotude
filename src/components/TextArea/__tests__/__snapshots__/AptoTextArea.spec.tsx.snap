// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AptoTextArea matches snapshot 1`] = `
<DocumentFragment>
  <textarea
    rows="3"
  />
</DocumentFragment>
`;
